docs/attestation.md

```diff
@@ -0,0 +1,120 @@
+
+    KEYS=( auditor )
+
+    for key in "${KEYS[@]}"; do
+      if [[ ! -e "$key".pk8 ]]; then
+        echo "Generating $key key"
+        # make_key exits with unsuccessful code 1 instead of 0
+        make_key "$key" "/CN=Robotnix auditor/" && exit 1
+      else
+        echo "Skipping generating $key since it is already exists"
+      fi
+    done
+    ```
+    Run the script to generate the keys:
+    ```bash
+    $ cd keys/
+    $ bash ../auditor-keys
+    ```
+
+ 2. Now you can enable the Auditor app in the configuration:
+    ```nix
+    {
+      keyStorePath = "/dev/shm/android-keys";
+      signing.enable = true;
+
+      apps = {
+        auditor.enable = true;
+        auditor.domain = "attestation.example.com";
+      };
+    }
+    ```
+    You also need to have signing enabled during build time.  Trying to build
+    for later signing with the release script doesn't build. (TODO: Probably a bug)
+    ```
+    Can't open /nix/store/a791pqlmh0npazbd6g310ga3b02mpz78-build=make-patched/target/product/security/auditor.x509.pem for reading, No such file or directory
+    140737348248832:error:02001002:system library:fopen:No such file or directory:crypto/bio/bss_file.c:69:fopen('/nix/store/a791pqlmh0npazbd6g310ga3b02mpz78-build=make-patched/target/product/security/auditor.x509.pem','r')
+    140737348248832:error:2006D080:BIO routines:BIO_new_file:no such file:crypto/bio/bss_file.c:76:
+    unable to load certificate
+    ```
+
+ 3. That's it from the Android side.  Note that the custom Auditor app will be
+    named “Robotnix Auditor”.  When you build GrapheneOS the normal Auditor app
+    will still be there, don't get confused (like I did).
+
+## Server side
+
+ 1. Before we begin we have to obtain the fingerprint of the custom Auditor app
+    and the AVB fingerprint.  To get the Auditor app fingerprint, we simply use
+    OpenSSL to extract the fingerprint of the signing certificate:
+    ```console
+    $ openssl x509 -noout -fingerprint -sha256 -in keys/auditor.x509.pem | awk -F '=' '{gsub(/:/,""); print $2}'
+    AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+    ```
+    The AVB fingerprint is a bit more tricky.  I own a Pixel 4a (sunfish) and
+    on this device the AVB fingerprint is simply the SHA256 hash of the AVB
+    key, but this is not the case on other devices.  Check the Auditor source
+    code for details.
+    ```console
+    $ sha256sum keys/sunfish/avb_pkmd.bin | awk '{print toupper($1)}'
+    BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB
+    ```
+
+ 2. Now you can import Robotnix in your NixOS configuration with the
+    aforementioned fingerprints.
+    ```nix
+    { config, lib, pkgs, ... }:
+    {
+      imports = [
+        ((builtins.fetchTarball {
+          name = "robotnix";
+          url =
+            "https://github.com/danielfullmer/robotnix/archive/master.tar.gz";
```

Comment:
Add a particular (dummy) commit here, instead of just `master.tar.gz`, since the sha256 will change whenever I push commits.